gulpfile.js

```diff
@@ -30,107 +30,104 @@ gulp.task('clean-tmp', function(cb) {
   del(['tmp'], cb);
 });
 
-// Send a notification when JSHint fails,
-// so that you know your changes didn't build
-function jshintNotify(file) {
-  if (!file.jshint) { return; }
-  return file.jshint.success ? false : 'JSHint failed';
-}
+const onError = $.notify.onError('Error: <%= error.message %>');
 
-function jscsNotify(file) {
-  if (!file.jscs) { return; }
-  return file.jscs.success ? false : 'JSRC failed';
+function lint(files) {
+  return gulp.src(files)
+    .pipe($.plumber(onError))
+    .pipe($.eslint())
+    .pipe($.eslint.format())
+    .pipe($.eslint.failOnError());
 }
 
 // Lint our source code
 gulp.task('lint-src', function() {
-  return gulp.src(['src/**/*.js'])
-    .pipe($.jshint())
-    .pipe($.jshint.reporter('jshint-stylish'))
-    .pipe($.notify(jshintNotify))
-    .pipe($.jscs())
-    .pipe($.notify(jscsNotify))
-    .pipe($.jshint.reporter('fail'));
+  return lint(['src/**/*.js']);
 });
 
 // Lint our test code
 gulp.task('lint-test', function() {
-  return gulp.src(['test/**/*.js'])
-    .pipe($.jshint())
-    .pipe($.jshint.reporter('jshint-stylish'))
-    .pipe($.notify(jshintNotify))
-    .pipe($.jscs())
-    .pipe($.notify(jscsNotify))
-    .pipe($.jshint.reporter('fail'));
+  return lint(['test/**/*.js']);
 });
 
 function getBanner() {
-  var banner = ['/**',
-    ' * <%= name %> - <%= description %>',
-    ' * @version v<%= version %>',
-    ' * @link <%= homepage %>',
-    ' * @license <%= license %>',
+  const banner = ['/**',
+    ` * ${ manifest.name } - ${ manifest.description }`,
+    ` * @version v${ manifest.version }`,
+    ` * @link ${ manifest.homepage }`,
+    ` * @license ${ manifest.license }`,
     ' */',
     ''].join('\n');
 
-  return _.template(banner)(manifest);
+  return banner;
 }
 
-function _build(entryFileName, destFolder, expFileName, expVarName, umd){
-  return esperanto.bundle({
-    base: 'src',
-    entry: entryFileName,
-    transform: function(source) {
-      var js_source = _.template(source)(manifest);
-
-      // Poor way of modifying dependency for modular build
-      if(!umd){
-        return js_source.replace('./state-class', 'marionette.toolkit.state-class');
-      }
-
-      return js_source;
+function _generate(bundle) {
+  const intro = getBanner();
+
+  return bundle.generate({
+    format: 'umd',
+    moduleName: config.exportVarName,
```

Comment:
we should pass in the moduleName as for the packages it should be using the `exportVarName`